### PR TITLE
[Merged by Bors] - feat: port Algebra.Module.Submodule.Lattice

### PR DESCRIPTION
**Pull request by int-y1**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/Submodule/Lattice.lean [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+  sum_mem fun i _ ↦ mem_supᵢ_of_mem i (h i)
+#align submodule.sum_mem_supr Submodule.sum_mem_supᵢ
+
+theorem sum_mem_bsupr {ι : Type _} {s : Finset ι} {f : ι → M} {p : ι → Submodule R M}
+    (h : ∀ i ∈ s, f i ∈ p i) : (∑ i in s, f i) ∈ ⨆ i ∈ s, p i :=
+  sum_mem fun i hi ↦ mem_supᵢ_of_mem i <| mem_supᵢ_of_mem hi (h i hi)
+#align submodule.sum_mem_bsupr Submodule.sum_mem_bsupr
+
+-- Porting note: Outdated comment? Lean 3 doesn't have this lemma in this file.
+/-! Note that `Submodule.mem_supr` is provided in `LinearAlgebra/Basic.lean`. -/
```
Comment (jcommelin) [inline Mathlib/Algebra/Module/Submodule/Lattice.lean:319]:
```suggestion
/-! Note that `Submodule.mem_supᵢ` is provided in `LinearAlgebra/Basic.lean`. -/
```
Does this make the comment less outdated?

**Review by jcommelin — COMMENTED** [review]
1 comment.
Otherwise, LGTM

File: Mathlib/Algebra/Module/Submodule/Lattice.lean (hunk 1 above)
Comment (int-y1) [reply Mathlib/Algebra/Module/Submodule/Lattice.lean:319]: Found it. Looks like `mem_supr` was refactored into a different file.
https://github.com/leanprover-community/mathlib/blob/fac369018417f980cec5fcdafc766a69f88d8cfe/src/linear_algebra/span.lean#L640

Comment (int-y1) [reply Mathlib/Algebra/Module/Submodule/Lattice.lean]: Updated `Basic.lean` to `Span.lean`.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**